### PR TITLE
new spec file and instructions for fedora + rpm

### PR DESCRIPTION
**Pull request by step21**
As mentioned here: https://github.com/open-eid/linux-installer/issues/18 I wanted to contribute the new spec file, plus instructions on how to keep it current (as much as that is possible) plus which process to follow for inclusion into fedora.
For now these instructions are in a readme.md file, but please let me know if you would prefer them on the wiki.

### CODE REVIEW

**Comment by metsma** [comment]
Can you remove config files and update readme to download latest from source location.
https://github.com/open-eid/qesteidutil/wiki/DeveloperTips#building

**Comment by step21** [comment]
Ok, should be better now. I also cleaned up the commits.

File: fedora/qesteidutil.spec [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+BuildRequires:  qtsingleapplication-qt5-devel
+Requires:       hicolor-icon-theme
+Requires:       opensc%{?_isa}
+Requires:       pcsc-lite-ccid%{?_isa}
+
+%description
+QEsteidUtil is an application for managing Estonian ID Card. In an
+user-friendly interface it is possible to change and unlock PINs, examine
+detailed information about personal data file on the card, extract and view
+certificates, set up mobile ID, and configure @eesti.ee email.
```
Comment (metsma) [inline fedora/qesteidutil.spec]: "set up mobile ID, "
This has been removed upstream

File: fedora/qesteidutil.spec [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+
+# Remove bundled qtsingleapplication to make sure it isn't used
+rm -rf qtsingleapplication
+# COPING is not needed as it is the license for the removed qtsingleapplication
+# bundle. See https://bugzilla.redhat.com/show_bug.cgi?id=1519323#c2
+rm -rf COPING
+
+%build
+
+%{cmake} . -DBREAKPAD=FALSE
```
Comment (metsma) [inline fedora/qesteidutil.spec]: "-DBREAKPAD=FALSE"
Has been removed upstream

File: fedora/qesteidutil.spec [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+
+BuildRequires:  cmake
+BuildRequires:  desktop-file-utils
+BuildRequires:  pkgconfig(openssl)
+BuildRequires:  pkgconfig(Qt5Designer)
+BuildRequires:  pkgconfig(libpcsclite) >= 1.7
+BuildRequires:  libappstream-glib
+BuildRequires:  qtsingleapplication-qt5-devel
+Requires:       hicolor-icon-theme
+Requires:       opensc%{?_isa}
```
Comment (metsma) [inline fedora/qesteidutil.spec]: opensc is not required by qesteidutil

File: fedora/qesteidutil.spec (hunk 2 above)
Comment (step21) [reply fedora/qesteidutil.spec]: Thanks for catching that. I think I had read about breakpad removal and would have thought it was already there but apparently not. Fixed now.

File: fedora/readme.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+# Guide
+## to create rpm package and update distribution package
+
+### General Information
+
+These instructions assume, that you have most/all of the tools needed for the compilation of software generally and qesteidutil specifically already installed. For the dependencies of qesteidutil, see the main Readme of the repository.
```
Comment (martinpaljak) [inline fedora/readme.md]: It does not hurt to give a copy-paste line for installing deps

File: fedora/qesteidutil.spec [hunk 5]
```diff
@@ -0,0 +1,225 @@
...
+License:        LGPLv2+
+URL:            https://github.com/open-eid/qesteidutil
+Source0:        %{url}/releases/download/v%{version}/%{name}-%{version}.tar.gz
+
+Source1:        config.json
+Source2:        config.pub
+Source3:        config.rsa
+Source4:        config.qrc
+
+Patch0:         sandbox-compilation.patch
```
[the remark this answers is not in the document]
Comment (step21) [reply fedora/qesteidutil.spec]: It is not needed anymore.

File: fedora/readme.md (hunk 4 above)
Comment (step21) [reply fedora/readme.md]: Should be there now.

**Comment by step21** [comment]
Is there anything left for me to do before this can be merged?

**Pull request merged**